docs/src/contributing/pull-request.md

```diff
@@ -0,0 +1,126 @@
+# The PR process for agave source code development
+
+## Preparation
+
+1. You’ll want a powerful machine for this. Building agave from sources takes a lot of RAM, make sure you have ~30GB to be sure it is enough.
+   1. Rust analyzer is quite capable of running on agave source tree, but expect high memory consumption
+2. Make a fork of anza-xyz/agave master branch under your own namespace in github. Let us assume it is called ubercoder/agave
+   1. Do not ever commit anything into the master branch of your fork. Keep it synced to upstream, this will make rebase process painless.
+
+## When not to file a PR
+
+1. Your code/comments may expose/highlight a security threat in the existing deployed instances
+
+
+## Procedure
+
+1. Make a feature branch on ubercoder/agave repo, e.g. fix\_all\_bugs
+2. Pull from ubercoder/agave the master and fix\_all\_bugs branches
+3. Make any necessary changes to fix\_all\_bugs
+   1. Commit and push as appropriate, make sure your commits are sensibly sized
+4. To keep track of changes in upstream anza-xyz/agave:
+   1. Sync the ubercoder/agave master branch with upstream. You can do this with cli or github. **This should always fast-forward** since you are never committing anything to  ubercoder/agave master.
+   2. Pull the latest version of ubercoder/agave master to your local machine if you have updated your fork via github
+   3. `git checkout fix\_all\_bugs; git rebase master;` to pull in the changes made in upstream’s master into your feature branch. Fix conflicts if necessary. Follow applicable rebase guides.
+   4. `git push –force** fix\_all\_bugs ` branch to rewrite its history (necessary due to rebase that was made)
+5. To create a PR:
+   1. Go to ubercoder/agave, switch to fix\_all\_bugs branch, and make PR against anza-xyz/agave master
+   2. Choose “make draft PR” option so it does not pollute the PR list before you are sure that CI checks pass
+   3. Make sure CI passes, then switch to actual PR for review
+   4. Find some nice reviewers to go over your code. If you do not know whom to choose as reviewer, ask around on discord.
+6. To appease the CI scripts, run this procedure before you push to the branch subject to CI:
```

Comment:
Maybe this section should come before the Create PR section?

Reply:
reorganized the sections a bit (prep first, CI prep second, finally actual PR and review)